test/unit-test/mqtt_agent_utest.c

```diff
@@ -853,3 +904,280 @@ void test_MQTTAgent_Terminate_success( void )
     TEST_ASSERT_NULL( command.pArgs );
     TEST_ASSERT_EQUAL_PTR( stubCompletionCallback, command.pCommandCompleteCallback );
 }
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior with invalid params.
+ */
+void test_MQTTAgent_CommandLoop_invalid_params( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+
+    mqttStatus = MQTTAgent_CommandLoop( NULL );
+
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    setupAgentContext( &mqttAgentContext );
+
+    mqttAgentContext.agentInterface.pMsgCtx = NULL;
+
+    mqttStatus = MQTTAgent_CommandLoop( NULL );
+
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior when sent command is NULL.
+ */
+void test_MQTTAgent_CommandLoop_sent_command_null( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTAgentCommandFuncReturns_t pReturnFlags;
+
+    setupAgentContext( &mqttAgentContext );
+
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    pReturnFlags.addAcknowledgment = false;
+    pReturnFlags.runProcessLoop = true;
+    pReturnFlags.endLoop = true;
+
+    MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_ProcessLoop_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTT_ProcessLoop_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior when sent command is not NULL.
+ */
+void test_MQTTAgent_CommandLoop_with_sent_command( void )
```

Comment:
Can you use a better name and description? The name and description was not helping me understand what the test was doing